ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/CommandStoreTest.java

```diff
@@ -155,84 +192,58 @@ public void shouldCleanupCommandStatusOnProduceError() {
     // When:
     commandStore.enqueueCommand(statementText, statement, KSQL_CONFIG, OVERRIDE_PROPERTIES);
 
-    // Then:
-    // main condition being verified is that the above call doesn't throw
-    verify(commandProducer);
+    // Then: condition being verified is that the above call doesn't throw
   }
 
   @Test
-  public void shouldEnqueueNewAfterHandlingExistingCommand() throws InterruptedException, ExecutionException {
-    final CommandStore commandStore = createCommandStoreThatAssignsSameId(commandId);
-
+  public void shouldEnqueueNewAfterHandlingExistingCommand() {
     // Given:
-    setupConsumerToReturnCommand(commandId, command);
-    expect(commandProducer.send(anyObject(ProducerRecord.class))).andAnswer(
-        () -> {
-          commandStore.getNewCommands();
-          return future;
-        }
-    ).times(1);
-    expect(commandProducer.send(anyObject(ProducerRecord.class))).andReturn(future);
-    future.get();
-    expectLastCall().andStubReturn(null);
-    replay(future, commandProducer);
     commandStore.enqueueCommand(statementText, statement, KSQL_CONFIG, OVERRIDE_PROPERTIES);
+    commandStore.getNewCommands();
 
     // When:
     commandStore.enqueueCommand(statementText, statement, KSQL_CONFIG, OVERRIDE_PROPERTIES);
 
-    // Then:
-    verify(future, commandProducer, commandConsumer);
+    // Then: condition being verified is that the above call doesn't throw
   }
 
   @Test
-  public void shouldRegisterBeforeDistributeAndReturnStatusOnGetNewCommands()
-      throws ExecutionException, InterruptedException {
-    final CommandStore commandStore = createCommandStoreThatAssignsSameId(commandId);
-
+  public void shouldRegisterBeforeDistributeAndReturnStatusOnGetNewCommands() {
```

Comment:
The intent of this test wasn't clear in the original version and isn't clear now. Not really your fault. However, while you're in here.... :D

So what its testing is that the `CommandStore`'s `commandStatusMap` is updated _before_ `send` is called on the producer, as otherwise there is a race condition.   To test that externally we need to be able to call `getNewCommands` before `send` returns and we need it to grab the status from the map, which means we need the consumer to return a record with a matching command Id to the one we're producing.  This is the bit that's missing.  I would suggest:

```java
@Test
  public void shouldRegisterBeforeDistributeAndReturnStatusOnGetNewCommands() {
    // Given:
    when(commandProducer.send(any(ProducerRecord.class))).thenAnswer(
        invocation -> {
          // Set up consumer to return record with matching key, i.e. force the race condition:
          final ProducerRecord<CommandId, Command> record = invocation.getArgument(0);
          when(commandConsumer.poll(any())).thenReturn(buildRecords(record.key(), record.value()));

          final QueuedCommand queuedCommand = commandStore.getNewCommands().get(0);
          assertThat(queuedCommand.getStatus().isPresent(), equalTo(true));
          assertThat(
              queuedCommand.getStatus().get().getStatus().getStatus(),
              equalTo(CommandStatus.Status.QUEUED));
          return recordMetadataFuture;
        }
    );

    // When:
    commandStore.enqueueCommand(statementText, statement, KSQL_CONFIG, OVERRIDE_PROPERTIES);

    // Then:
    // verifying the commandProducer also verifies the assertions in its Answer were run
    verify(commandProducer).send(any(ProducerRecord.class));
```